pkg/test/framework/components/namespace/kube.go

```diff
@@ -182,14 +182,24 @@ func newKube(ctx resource.Context, nsConfig *Config) (Instance, error) {
 }
 
 // createNamespaceLabels will take a namespace config and generate the proper k8s labels
-func createNamespaceLabels(cfg *Config) map[string]string {
+func createNamespaceLabels(ctx resource.Context, cfg *Config) map[string]string {
 	l := make(map[string]string)
 	l["istio-testing"] = "istio-test"
 	if cfg.Inject {
-		if cfg.Revision != "" {
-			l[label.IoIstioRev.Name] = cfg.Revision
-		} else {
-			l["istio-injection"] = "enabled"
+		// do not add namespace labels when dealing with multiple revisions since
```

Comment:
Not sure I follow this comment ... can you expand it a bit?

Reply:
So the way our injection labels work today, object selectors get ignored if there's a namespace selector present. 

This means when a namespace has `cfg.Inject=true` and we're using object selectors, we should leave it alone since the deployments will give the pods `istio.io/rev=foo`, and when a namespace has `cfg.Inject=false` and we're using object selectors, we need to explicitly disable injection with `istio-injection=disabled`.